packages/beacon-node/src/chain/chain.ts

```diff
@@ -479,6 +479,12 @@ export class BeaconChain implements IBeaconChain {
     this.logger.verbose("Clock slot", {slot});
 
     // CRITICAL UPDATE
+    try {
+      this.forkChoice.checkSaneForkChoice();
+    } catch (e) {
+      this.logger.error("Critical error in forkChoice, shutting down gracefully", {}, e as Error);
+      process.kill(process.pid, "SIGINT");
```

Comment:
No you **must** never call process.kill outside of the cli package. This code must be consumed programatically

Reply:
See

https://github.com/ChainSafe/lodestar/blob/20edb58ca3207b7cf8e1450172f9575cd1defaea/packages/cli/src/cmds/validator/handler.ts#L45